src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1234 @@
+
+            # Output weights.
+            if not self.config.tie_word_embeddings:
+                _init_modernbert_weights(self.config, self.decoder, module_type=ModernBertModuleType.out_module)
+
+    def get_output_embeddings(self):
+        return self.decoder
+
+    def set_output_embeddings(self, new_embeddings: nn.Linear):
+        self.decoder = new_embeddings
+
+    @torch.compile(dynamic=True)
+    def compiled_head(self, output: torch.Tensor) -> torch.Tensor:
+        return self.decoder(self.head(output))
+
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor],
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.Tensor] = None,
+        labels: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = None,
+        indices: Optional[torch.Tensor] = None,
+        cu_seqlens: Optional[torch.Tensor] = None,
+        max_seqlen: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        seq_len: Optional[int] = None,
+        **kwargs,
+    ) -> Union[Tuple[torch.Tensor], MaskedLMOutput]:
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if self.config.unpad_inputs:
+            if indices is None and cu_seqlens is None and max_seqlen is None:
+                batch_size, seq_len = input_ids.shape[:2]
+                if self.config.unpad_no_grad:
+                    input_ids, indices, cu_seqlens, max_seqlen, position_ids, labels = self._unpad_inputs_no_grad(
+                        input_ids=input_ids, attention_mask=attention_mask, position_ids=position_ids, labels=labels
+                    )
+                else:
+                    input_ids, indices, cu_seqlens, max_seqlen, position_ids, labels = self._unpad_inputs(
+                        input_ids=input_ids, attention_mask=attention_mask, position_ids=position_ids, labels=labels
+                    )
+
+        output = self.model(
+            input_ids,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            indices=indices,
+            cu_seqlens=cu_seqlens,
+            max_seqlen=max_seqlen,
+            batch_size=batch_size,
+            seq_len=seq_len,
+            return_dict=return_dict,
+        )
+        output = output[0]
+
+        if self.sparse_prediction and labels is not None:
+            # flatten labels and output first
+            labels = labels.view(-1)
+            output = output.view(labels.shape[0], -1)
+
+            # then filter out the non-masked tokens
+            mask_tokens = labels != self.sparse_pred_ignore_index
+            output = output[mask_tokens]
+            labels = labels[mask_tokens]
+
+        logits = self.compiled_head(output)
+
+        loss = None
+        if labels is not None:
+            loss = self.loss_function(logits, labels, vocab_size=self.config.vocab_size)
```

Comment:
`self.loss_function` does not exist, I believe.

Reply:
it does! 🤗 the loss_utils was correctly editted!